### DISCUSSION TITLE
Payload raw data in hex

### DISCUSSION THREAD

**Discussion by Vrielmann-EE** [moscajs/aedes#749] [Q&A] [answered]
I would like to send raw hex values over the payload, to keep the amount of data very small. But if I send a value greater or equal than  0x80 - I am not recieving the value 0x80. In this case I am receiving "c2 80".

Thanks in advance

**Comment by robertsLando** [comment, marked as answer]
@Vrielmann-EE 
Could you show me how you are sending it? Could you paste here a snippet that reproduces the issue?

**Reply by Vrielmann-EE** [reply]
Dear Roberts,

thank you for your quick reply, here are a small picture
For my payload (on the left side) "0x80" - I am receiving  "194128" on the right side: 194 (0xC2) 128 (0x80)

![Unbenannt](https://user-images.githubusercontent.com/63099602/175515276-35b0e4d5-f4a5-45f4-a470-251f36cce564.PNG)

**Reply by robertsLando** [reply]
I have a feel that on the right side the `0x80` is treated as a stirng instead of a raw byte payload, try to send `€` (that should be 0x80 converted to hex)

**Reply by Vrielmann-EE** [reply]
For "0x7F" I am just receiving the value "0x7F" as raw payload "127"

**Reply by robertsLando** [reply]
Could you reproduce it with a simple nodejs program using aedes and mqttjs for the client?

**Reply by Vrielmann-EE** [reply]
I did - it works as expected. With this small test client on connecting to the same server, I am sending and receiving 128 (0x80), on the right side: Raw payload: 128:

```
const mqtt = require('mqtt')
const client  = mqtt.connect('mqtt://localhost')

client.on('connect', function () {
    client.subscribe('K2/ULN', function (err) {
        if (!err) {
            const buffer = new Uint8Array(1);
            buffer[0] = 0x80;
            client.publish('K2/ULN', buffer);
        }
    });
})

client.on('message', function (topic, message) {
    console.log(message);
    client.end();
})
```

My first test client, from which I had sent, is the MQTTbox: https://chrome.google.com/webstore/detail/mqttbox/kaajoficamnjijhkeomgfljpicifbkaf

**Reply by robertsLando** [reply]
@Vrielmann-EE Ok so it's not an aedes bug, it's a bug with MQTT Box I think

**Reply by Vrielmann-EE** [reply]
Ok, I think so too! MQTTbox is the only client I know where I can change the "payload type" - Would be grateful for any other hint. How can I mark this question as answered?

**Reply by robertsLando** [reply]
I did it now

**Reply by robertsLando** [reply]
I usually use mqtt explorer as mqtt client
